packages/office-ui-fabric-react/src/utilities/keytips/KeytipManager.ts

```diff
@@ -0,0 +1,430 @@
+      if (keytipProps.overflowSetSequence) {
+        keytipSequence = mergeOverflowKeySequences(keytipSequence, keytipProps.overflowSetSequence);
+      }
+
+      // Set currentKeytip
+      keytipTree.currentKeytip = keytipTree.getNode(convertSequencesToKeytipID(keytipSequence));
+      if (keytipTree.currentKeytip) {
+        // Show all children keytips if there
+        const children = keytipTree.getChildren();
+        children.length && this.showKeytips(children);
+
+        if (keytipTree.currentKeytip.onExecute) {
+          keytipTree.currentKeytip.onExecute(this._getKeytipDOMElement(keytipTree.currentKeytip.id));
+        }
+      }
+
+      // Unset _newCurrentKeytipSequences
+      this._newCurrentKeytipSequences = undefined;
+    }
+
+    return uniqueKeytip.uniqueID;
+  }
+
+  /**
+   * Register a persisted keytip
+   * This means just adding a KeytipTreeNode
+   *
+   * @param keytipProps - Persisted Keytip to register
+   * @returns {string} Unique ID for this persisted keytip
+   */
+  public registerPersistedKeytip(keytipProps: IKeytipProps): string {
+    const uniqueID = this._getNextUniqueID();
+    this.keytipTree.addNode(keytipProps, uniqueID, true);
+    return uniqueID;
+  }
+
+  /**
+   * Update a keytip's props
+   *
+   * @param keytipProps - Keytip to update
+   * @param uniqueID - Unique ID of this keytip
+   */
+  public updateKeytip(keytipProps: IKeytipProps, uniqueID: string): void {
+    // Update keytip in this.keytips
+    const uniqueKeytip = this._constructUniqueKeytip(keytipProps, uniqueID);
+    const keytipIndex = this._findKeytipIndex(uniqueKeytip);
+    if (keytipIndex >= 0) {
+      this._updateKeytip(uniqueKeytip, keytipIndex);
+    }
+  }
+
+  /**
+   * Unregisters a keytip
+   *
+   * @param keytipToRemove - IKeytipProps of the keytip to remove
+   */
+  public unregisterKeytip(keytipToRemove: IKeytipProps, uniqueID: string): void {
+    // Add the unique ID to the keytip
+    const uniqueKeytipToRemove = this._constructUniqueKeytip(keytipToRemove, uniqueID);
+
+    // Remove keytipToRemove from this.keytips
+    this.keytips = this.keytips.filter((uniqueKtp: IUniqueKeytip) => {
+      return !(fullKeySequencesAreEqual(uniqueKtp.keytip.keySequences, keytipToRemove.keySequences) && uniqueKtp.uniqueID === uniqueID);
+    });
+
+    // Remove keytip from the delayed queue
+    this._removeKeytipFromQueue(convertSequencesToKeytipID(keytipToRemove.keySequences));
+
+    // Remove the node from the Tree
+    this.keytipTree.removeNode(keytipToRemove, uniqueID);
+    this._layer && this._layer.setKeytips(this.getKeytips());
+  }
+
+  /**
+   * Unegister a persisted keytip
+   * This means just removing it from the KeytipTree
+   *
+   * @param keytipToRemove - Persisted keytip to remove
+   * @param uniqueID - Unique ID of this persisted keytip
+   */
+  public unregisterPersistedKeytip(keytipToRemove: IKeytipProps, uniqueID: string): void {
+    // Add the unique ID to the keytip
```

Comment:
>// Add the unique ID to the keytip [](start = 4, length = 34)

[nit] remove keytip from keytipTree, or remove comment altogether

Reply:
Oh yeah that comment looks like it was copy-pasted :)